### ISSUE TITLE
Enhancement. Important for my data use case.

### ISSUE THREAD

**Issue by johnpittman** [sanusart/react-dropdown-select#17]
Can you please add callbacks as options for valueField/labelField to access data? We use nested data structure so need something like: labelField={(item) => { return item.attributes.name;}}

Currently you're just assuming data structure are flat.

**Comment by sanusart** [comment]
@johnpittman feels like an age case. Can't you just re-map the options to the desired structure before you pass it to `options` prop? Something like the following should do the job:

```javascript
const newOptions = () =>
    oldOptions.map(item => ({
      label: item.attributes.name,
      value: item.email
    }));

...

return (
  <Select
     options={newOptions}
     onChange={() => null}
  />
);
```

**Comment by johnpittman** [comment]
It's a very common feature. All other components I use or build include the
ability. We have filters with 10 dropdowns that all have large datasets.
The reason for not remapping each is because a lot of dropdown have over
1000 options as the data set so you doing double the work/bloating in the
UI thread just to read the label fields.

John R. Pittman |
<https://plus.google.com/u/0/100394734061909211307/about/p/pub>

johnrichardpittman@gmail.com

mobile: 561-460-2051

**Comment by sanusart** [comment]
I'll check the impact of your suggestion on the current codebase. I am not
sure it is less load on UI thread as select use this fields pretty heavily
internally to access data and work on dataset while simple looping,
memoized/cached is "cheaper".

**Comment by sanusart** [comment]
@johnpittman So this is very similar to original request, you can specify path instead of a string to access nested property. e.g.
```
<Select 
  valueField="attributes.name"
  labelField="attributes.name" 
  options={options}
  onChange={() => null}
/>
```

It is significant change to the code base and I want to keep testing it a bit more, so gonna keep it in the branch for some time. You may check it from the branch on your project if you want with loading it from branch: 

`npm uninstall --save react-dropdown-select && npm uninstall --save react-dropdown-select#experimental-suppot-for-dotted-path-to-access-labelFiels-and-valueField`

**Comment by sanusart** [comment]
Available since v3.2.0 
Example of usage [here](https://github.com/sanusart/react-dropdown-select/blob/master/docs/src/examples/ExternalClear.js)
